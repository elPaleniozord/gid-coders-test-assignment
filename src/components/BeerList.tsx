import React, { useContext, useEffect, useMemo, useState } from "react"
import { PunkAPIContext } from "../ctx/PunkAPIContext"
import BeerItem from "./BeerItem"
import Loader from "./Loader"

const BeerList = () => {
  const {loading, beers, fetchMore, favorites} = useContext(PunkAPIContext)

  const [beerList, setBeerList] = useState<(Element | JSX.Element)[]>([])

  const handlePagination = () => {
    fetchMore && fetchMore()
  }
  
  useEffect(() => {
    const from = beerList.length
    const update = []
     
    for(let i=from; i<beers.length; i++) {
      update.push((<BeerItem key={beers[i].id} {...beers[i]} />))
    }

    setBeerList([...beerList, ...update])
  }, [beers])

  return (
    <div>
      <ul aria-label='beer-list'>
        {beerList}
      </ul>
      {loading && (<Loader />)}
      <button onClick={handlePagination}>More Beers!</button>
    </div>

  )
}

export default BeerList