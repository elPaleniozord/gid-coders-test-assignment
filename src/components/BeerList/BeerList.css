.list-container {
  margin-top: 1.2rem;
  padding-bottom: 2.4rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
}

.beer-list {
  list-style: none;
  padding: 0;
}

.list-button {
  min-width: calc(320px - .6rem);
  font-family: Veneer;
  height: 44px;
  background: var(--highlight);
  border: none;
  border-radius: 5px;
  color: var(--offblack);
  font-size: 1.5rem;
}
.list-button:hover {
  box-shadow: inset 0px 0px 10px rgb(0, 102, 131);
}
.list-button:active {
  background: rgb(0, 138, 177);
}