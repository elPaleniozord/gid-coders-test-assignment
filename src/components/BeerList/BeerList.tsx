import React, { useContext, useEffect, useState } from "react"
import { PunkAPIContext } from "../../ctx/PunkAPIContext"
import BeerItem from "../BeerItem/BeerItem"
import Loader from "../Loader/Loader"

import './BeerList.css'

const BeerList = () => {
  const {loading, beers, fetchMore, favorites} = useContext(PunkAPIContext)

  const [beerList, setBeerList] = useState<(Element | JSX.Element)[]>([])

  const handlePagination = () => {
    fetchMore && fetchMore()
  }
  
  useEffect(() => {
    const from = beerList.length
    const update = []
     
    for(let i=from; i<beers.length; i++) {
      update.push((<BeerItem key={beers[i].id} {...beers[i]} />))
    }

    setBeerList([...beerList, ...update])
  }, [beers])

  return (
    <div className="list-container">
      <ul className='beer-list' aria-label='beer-list'>
        {beerList}
      </ul>
      {!loading && (<Loader />)}
      <button className='list-button_fetch' onClick={handlePagination}>More Beers!</button>
    </div>
  )
}

export default BeerList