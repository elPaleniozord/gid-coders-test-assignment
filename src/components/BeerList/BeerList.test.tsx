import { render } from '@testing-library/react';
import BeerList from "./BeerList"

test('should render Favorites Component properly', () => {
  const { container } = render(<BeerList />)
  const list = container.querySelector('div')
  expect(list).toMatchSnapshot()
})