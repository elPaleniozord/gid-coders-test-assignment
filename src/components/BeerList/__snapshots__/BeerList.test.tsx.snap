// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Favorites Component properly 1`] = `
<div
  class="list-container"
>
  <ul
    aria-label="beer-list"
    class="beer-list"
  />
  <button
    class="list-button list-button_fetch"
  >
    More Beers!
  </button>
</div>
`;
